app/src/main/java/com/android/unio/ui/home/Home.kt

```diff
@@ -102,152 +90,199 @@ fun HomeScreen(
-                                else MaterialTheme.colorScheme.inversePrimary,
-                            modifier =
-                                Modifier.clickable {
-                                      selectedTab = "All"
-                                      coroutineScope.launch {
-                                        animatablePosition.animateTo(
-                                            0f, // Starting point for "All"
-                                            animationSpec =
-                                                tween(durationMillis = 1000) // Animation duration
-                                            )
-                                      }
-                                    }
-                                    .padding(horizontal = 16.dp)
-                                    .onGloballyPositioned { coordinates ->
-                                      allTabWidth = (coordinates.size.width / density).dp
-                                      allTabXCoordinate = coordinates.positionInRoot().x
-                                    }
-                                    .testTag(HomeTestTags.TAB_ALL))
+        TopBar(
+            navigationAction,
+            searchState,
+            searchResults,
+            userViewModel,
+            searchViewModel,
+            eventViewModel,
+            paddingValues)
+      })
+}
 
-                        // Clickable text for "Following"
-                        Text(
-                            text = "Following",
-                            color =
-                                if (selectedTab == "Following") MaterialTheme.colorScheme.primary
-                                else MaterialTheme.colorScheme.inversePrimary,
-                            modifier =
-                                Modifier.clickable {
-                                      selectedTab = "Following"
-                                      coroutineScope.launch {
-                                        animatablePosition.animateTo(
-                                            1f, // Ending point for "Following"
-                                            animationSpec =
-                                                tween(durationMillis = 1000) // Animation duration
-                                            )
-                                      }
-                                    }
-                                    .padding(horizontal = 16.dp)
-                                    .onGloballyPositioned { coordinates ->
-                                      followingTabWidth = (coordinates.size.width / density).dp
-                                      followingTabXCoordinate = coordinates.positionInRoot().x
-                                    }
-                                    .testTag(HomeTestTags.TAB_SAVED))
-                      }
+@Composable
+fun HomeContent(
```

Comment:
After discussing this IRL, we concluded that it would cause problems to do this so I will leave the View models for now. Same for the other comments saying to use hilt.